src/proxyd/proxyd_launcher.star

```diff
@@ -24,10 +24,13 @@ CONFIG_TEMPLATE_FILEPATH = "{0}/{1}.tmpl".format(TEMPLATES_FILEPATH, CONFIG_FILE
 CONFIG_DIRPATH_ON_SERVICE = "/etc/proxyd"
 
 
-def get_used_ports():
+def get_used_ports(proxyd_params):
+    rpc_port = (
+        proxyd_params.rpc_port if hasattr(proxyd_params, "rpc_port") else HTTP_PORT_NUM
```

Comment:
I'm not sure I understand the goal here. Why would we want to customize the port numbers?

Also, even if we do, I see that pattern repeated throughout the file, testing if proxyd_params has some field. Wouldn't be easier to define those fields in the default_proxyd_params? this way we would know for sure that there's a value to r ead.